apps/meteor/client/views/room/components/body/composer/messageBox/MessageBox.tsx

```diff
@@ -332,17 +332,21 @@ const MessageBox = ({
 		select,
 		commandsRef,
 		callbackRef: c,
+		filter
 	} = useComposerBoxPopup<{ _id: string; sort?: number }>({
 		configurations: composerPopupConfig,
 	});
 
 	const mergedRefs = useMessageComposerMergedRefs(c, textareaRef, callbackRef, autofocusRef);
 
+	const showComposerBoxPopupPreview = popup && !popup.preview;
+	const showComposerBoxPopup = showComposerBoxPopupPreview && popup?.triggerLength ? typeof filter === 'string' && popup.triggerLength - 1 < filter.length : true;
```

Comment:
```suggestion
	const showComposerBoxPopup = shouldPopupPreview && popup?.triggerLength ? typeof filter === 'string' && popup.triggerLength - 1 < filter.length : true;
```

Reply:
Additionally: what do you think about creating a helper function to isolate it and reduce cognitive load to understand the code

Reply:
There are some linting issues also!